opendevin/sandbox/docker/ssh_box.py

```diff
@@ -169,7 +169,7 @@ def setup_user(self):
 
     def start_ssh_session(self):
         # start ssh session at the background
-        self.ssh = pxssh.pxssh()
+        self.ssh = pxssh.pxssh(echo=False)
```

Comment:
@xingyaoww wdyt, does this fix the "weird behavior" you saw?

Reply:
I think this just prevents the command from being returned. it might break the way how `ssh_box.sh` currently finds the response. 

The issue is i was seeing is that, the first time you call `.ssh.before`, the return value you get *might not* be complete: you only get a part of the output. And in the next round, when you try to issue `echo $?` to get the exit code, you will get the other part of the output of previous command along with the exit code, which is highly undesirable.

Reply:
This PR also dramatically simplifies the response-finding logic, all it does is return the text found .before the next cmd prompt match.

Do you have an example situation that can cause the return data to extend past the next cmd prompt match? I have some unit tests that I ran locally and couldn't easily come up with such a scenario.

Reply:
that's why I call it "weird" in the sense that it is not easily reproducible (sometime happens, sometimes works) :(
But anyway, we can go with your current logic for now and try to fix it in the future if that issue comes up again.